### PR TITLE
WASM fixes - Fix #12194

### LINKED ISSUE #12194: Symbolnames not working in RBin.wasm
screenshot 2018-11-16 at 12 18 02

[main.wasm.zip](https://github.com/radare/radare2/files/2589054/main.wasm.zip)

### PR DESCRIPTION
**Pull request by wargio**
Fixes for fix #12194

### CODE REVIEW

File: libr/bin/p/bin_wasm.c [hunk 1]
```diff
@@ -216,6 +214,8 @@ static RList *imports(RBinFile *bf) {
 		if (!(ptr = R_NEW0 (RBinImport))) {
 			goto bad_alloc;
 		}
+		eprintf("%s\n", import->field_str);
```
Comment (radare) [inline libr/bin/p/bin_wasm.c]: debug printfs

Comment (ret2libc) [reply libr/bin/p/bin_wasm.c]: you can now use `R_LOG_DEBUG` if you want :)

Comment (wargio) [reply libr/bin/p/bin_wasm.c]: removed

File: libr/bin/format/wasm/wasm.h [hunk 2]
```diff
@@ -215,6 +222,7 @@ RList *r_bin_wasm_get_globals (RBinWasmObj *bin);
 RList *r_bin_wasm_get_elements (RBinWasmObj *bin);
 RList *r_bin_wasm_get_codes (RBinWasmObj *bin);
 RList *r_bin_wasm_get_datas (RBinWasmObj *bin);
+RList *r_bin_wasm_get_symtab (RBinWasmObj *bin);
```
Comment (radare) [inline libr/bin/format/wasm/wasm.h]: no space before ( on the function signatures

File: libr/bin/p/bin_wasm.c [hunk 3]
```diff
@@ -13,24 +13,29 @@ static bool check_bytes(const ut8 *buf, ut64 length) {
...
 
-static bool load_bytes(RBinFile *bf, void **bin_obj, const ut8 *buf, ut64 sz, ut64 loadaddr, Sdb *sdb){
-	if (!buf || !sz || sz == UT64_MAX) {
-		return false;
-	}
-	if (!check_bytes (buf, sz)) {
-		return false;
+static bool check_bytes_buf(RBuffer* rbuf) {
+	ut8 buf[4] = {0};
+	return rbuf && r_buf_read_at(rbuf, R_BUF_CUR, buf, 4) == 4 && !memcmp (buf, R_BIN_WASM_MAGIC_BYTES, 4);
```
Comment (radare) [inline libr/bin/p/bin_wasm.c]: missing space before (

File: libr/bin/p/bin_wasm.c [hunk 4]
```diff
@@ -13,24 +13,29 @@ static bool check_bytes(const ut8 *buf, ut64 length) {
...
+}
+
+static int find_symbol(const ut8 *p, const RBinWasmSymbol* q) {
+	return q->id != (*p);
+}
+
+static void * load_buffer(RBinFile *bf, RBuffer *buf, ut64 loadaddr, Sdb *sdb) {
+	r_return_val_if_fail (bf || buf || r_buf_size (buf) == UT64_MAX, NULL);
```
Comment (radare) [inline libr/bin/p/bin_wasm.c]: r_return_val_if_fail must use && instead of  ||

**Comment by wargio** [comment]
pls squash

**Pull request merged**